src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/extensions/BitbucketPostBuildStatus.java

```diff
@@ -0,0 +1,57 @@
+package com.atlassian.bitbucket.jenkins.internal.scm.extensions;
+
+import com.atlassian.bitbucket.jenkins.internal.provider.DefaultJenkinsProvider;
+import com.atlassian.bitbucket.jenkins.internal.provider.JenkinsProvider;
+import com.atlassian.bitbucket.jenkins.internal.status.BitbucketRevisionAction;
+import com.atlassian.bitbucket.jenkins.internal.status.BuildStatusPoster;
+import com.google.inject.Injector;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.util.BuildData;
+import org.jenkinsci.plugins.gitclient.CheckoutCommand;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+/**
+ * Although this class extends the GitSCMExtension, it is only intended for use with the BitbucketSCM, which instantiates
```

Comment:
If this is only intended to use with BitbucketSCM, you could move it in SCM package and change the visibility of this class as default.

Reply:
Is there a preference for one or the other? I'm pretty neutral.